test/clara_specs/core_test.clj

```diff
@@ -1,85 +1,75 @@
+           :docstring "Find the client representative and send a notification of a support request.",
+           :lhs [[:condition {:type SupportRequest, :constraints [(= ?client client)]}]
+                 [:condition {:type ClientRepresentative :constraints [(= ?client client) (= ?name name)]}]]
+           :implies =>
+           :rhs [(println "Notify" ?name "that" ?client "has a new support request!")]}))
+
   (is (= (s/conform ::clara/defrule-form
                     (rest '(defrule get-windspeeds-from-location
                                     [?temp <- [Temperature (= ?location location)]]
                                     [?wind <- (acc/all) :from [WindSpeed (= ?location location)]]
                                     =>
                                     ())))                   ;; do something
+         '{:name get-windspeeds-from-location,
+           :lhs [[:fact-binding-expr
+                  {:fact-binding {:binding-var ?temp, :sep <-}
+                   :condition {:type Temperature :constraints [(= ?location location)]}}]
+                 [:accumulator-expr
+                  {:result-binding ?wind,
+                   :<- <-
+                   :accumulator (acc/all)
+                   :from :from
+                   :condition {:type WindSpeed :constraints [(= ?location location)]}}]],
+           :implies =>
+           :rhs [()]})))
 
-         '{:name       get-windspeeds-from-location
-           :conditions [[:fact-expr
-                         {:binding   {:binding-var ?temp
-                                      :sep         <-}
-                          :fact-type [Temperature
-                                      (= ?location location)]}]
-                        [:accumulator
-                         {:binding-var ?wind
-                          :fact-expr   {:expressions [(= ?location location)]
-                                        :fact-type   WindSpeed}
-                          :fact-type   (acc/all)
-                          :_           :from
-                          :sep         <-}]]
-           :implies    =>
-           :rhs        [()]})))
 
 (deftest defquery-examples
   (is (= (s/conform ::clara/defquery-form
                     (rest '(defquery get-promotions
                                      "Query to find promotions for the purchase."
                                      [:?type]
-                                     [?promotion <- Promotion (= ?type type)])))
-         '{:name       get-promotions
-           :docstring  "Query to find promotions for the purchase."
-           :args       [:?type]
-           :conditions [[:fact-expr
-                         {:binding     {:binding-var ?promotion
+                                     [?promotion <- [Promotion (= ?type type)]])))
+         '{:name      get-promotions
+           :docstring "Query to find promotions for the purchase."
+           :args      [:?type]
+           :lhs       [[:fact-binding-expr
```

Comment:
@mrrodriguez Re: exprs vs. conditions from above, am I understanding correctly that `{:lhs :fact-binding-expr {:fact-binding ...}` should be `[lhs [[:condition {fact-binding ...}` ? 

If so, what do you suggest for the current `:condition` construct? (`:condition {:type Promotion :constraints [(= ?type type)]}`)

Reply:
Yes, I believe that is the naming change I was suggesting. I see where it is difficult to find the correct names for each of these pieces though. There is a bit of a mismatch perhaps in how to structure the spec vs the schema definitions that the names aligned with previously. The "condition" was the entire form from the LHS (that may or may not have a fact binding within it). The vector within the condition is the "constraints" part. The "fact type" part is just part of the condition.

So I think naming the "current `:condition` construct" is tricky since it didn't have a name on its own before basically.